change PhoneNumber and use it in ContactInformation

daschJulia opened the pull request:
only format phonenumber when getInternationalFormatOfPhoneNumber() is called.
do only format if the rawPhoneNumber of the PhoneNumber is not null and no
PhoneNumberFormattingException is thrown.
PhoneNumber is used in ContactInformation instead of Strings of a phone number.

daschJulia committed:
change PhoneNumber and use it in ContactInformation

only format phonenumber when getInternationalFormatOfPhoneNumber() is called.
do only format if the rawPhoneNumber of the PhoneNumber is not null and no
PhoneNumberFormattingException is thrown.
PhoneNumber is used in ContactInformation instead of Strings of a phone number.

rjayasinghe commented on pom.xml:
```diff
@@ -75,6 +75,11 @@
             <version>1.10.19</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Uhoh. I guess we'd like to stay without a dependency to Spring framework. If this has been added for the StringUtils use case please use something similar from e.g. apache-commons or guava etc.

rjayasinghe commented on src/main/java/net/contargo/types/contactinfo/normalization/PhoneNumberNormalizer.java:
```diff
@@ -11,16 +12,10 @@
     public Optional<String> normalizeNumber(String phoneNumber) {
 
-        if (phoneNumber == null) {
+        if (StringUtils.hasText(phoneNumber)) {
```
I guess this is the one. I had a look at the other code in this project. There are cases where we use StringUtils.isEmpty() from Apache Commons. Please adjust your imports. :)

rjayasinghe commented:
Looks good so far. Please adjust the Spring framework dependency. After that somebody can merge. :)

daschJulia committed:
remove spring boot dependency

avoid using many dependencies.
use other StringUtil class to check if phonenumber has
text.

daschJulia committed:
modify Contactinformation constructors

add second constructor to be able to set PhoneNumbers as Phonenumber or as String.

rjayasinghe reviewed (approved):
Looks good now. :)

daschJulia committed:
revert new constructer

for the moment do not add second constructor there are many apps using the primary one.
compiler does not now which one it should use if both numbers are null.

daschJulia committed:
change ValidPhoneNumberValidator

validator is changed to check für numbers with only zeros and
now uses the PhoneNumber object to Validate if it is a phone number.

daschJulia committed:
change getAreaCodeAndConnectionNumberFromFixedNumber(preFormattedNumber)

first element of the splitted preFormatted Number is country code, second element should be
the areacode if the preFormattedNumber does have more than 2 elements.
If there are only two elements then connection number is splittedPreformattedNumber[1]
and areacode does not exist.

rjayasinghe commented on src/main/java/net/contargo/types/telephony/PhoneNumber.java:
```diff
@@ -58,7 +65,11 @@ public void setPhoneExtension(String phoneExtension) {
     public Country getCountry() throws PhoneNumberFormattingException {
 
-        return new Country(phoneNumberFormatter.getRegionCode(rawPhoneNumber));
+        if (Objects.isNull(country)) {
```
I think a StringUtils.isBlank() would be better here.

daschJulia replied:
Country is an Object :)

daschJulia committed:
add more tests for phone numbers

test more special phonenumbers to be sure phone numbers are correctly formatted.
